src/Plugin/KeyProvider/PrivateFileKeyProvider.php

```diff
@@ -111,20 +141,26 @@ public function getKeyValue(KeyInterface $key) {
 
   /**
    * {@inheritdoc}
-   *
-   * @throws \Drupal\key\Exception\KeyValueNotSetException
-   *   Thrown when the key cannot be saved.
    */
   public function setKeyValue(KeyInterface $key, $key_value) {
+    try {
+      $this->checkRequirements($key);
+    }
+    catch (KeyProviderRequirementsException $exception) {
+      $context = [
+        '@message' => (string) $exception,
+      ];
+      $context += Error::decodeException($exception);
+      $this->logger->error('Could not save Apigee Edge authentication key value in the private file storage: @message %function (line %line of %file). <pre>@backtrace_string</pre>', $context);
+      return FALSE;
+    }
+
     $file_uri = $this->getFileUri($key);
     $file_path = dirname($file_uri);
     // Make sure the folder exists.
     file_prepare_directory($file_path, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS);
```

Comment:
Procedural functions from file.inc will be deprecated in Drupal 8.7.
Let's use successor file system service method calls instead of them everywhere it is possible.

https://github.com/drupal/core/blob/8.7.x/includes/file.inc#L317